docs/docs/user-guide/training-data-format.mdx

```diff
@@ -0,0 +1,659 @@
+nlu:
+- intent: greet
+  examples: |
+  - Hey
+  - Hi
+  - hey there [Sara](name)
+
+- intent: faq/language
+  examples: |
+  - What language do you speak?
+  - Do you only handle english?
+
+e2e_tests:
+- user: |
+    hey
+  intent: greet
+- action: utter_greet
+- user: |
+    what language do you speak
+  intent: faq/language
+- action: respond_faq
+```
+
+
+## NLU Training Data
+
+NLU training data consists of example user utterances categorized by
+**intent**, i.e. what the user is trying to convey or accomplish with their
+message. Training examples can also include **entities**. Entities are structured
+pieces of information that can be extracted from a user's message. The other components
+of NLU training data are there to help the model identify intent and entity
+correctly.
+
+NLU training data is defined under the `nlu` key. Items that can be added under this key are:
+
+- [Training examples](#training-examples) grouped by user intent,
+  optionally with annotated [entities](#entities)
+- [Synonyms](#synonyms)
+- [Regular Expression Features](#regular-expression-features)
+- [Lookup Tables](#lookup-tables)
+
+Here is a simple example of each item type:
+
+```yaml
+nlu:
+- intent: check_balance
+  examples: |
+  - What's my [credit](account) balance?
+  - What's the balance on my [credit card account]{"entity":"account","value":"credit"}
+
+- synonym: credit
+  examples: |
+  - credit card account
+  - credit account
+
+- regex: account_number
+  examples: |
+  - \d{10:12}
+
+- lookup: banks
+  examples: |
+  - JPMC
+  - Comerica
+  - Bank of America
+```
+
+
+### Training Examples
+
+Training examples are grouped by [intent](../glossary#intent) and listed under the
+`examples` key. Usually, you'll list one example per line as follows:
+
+```yaml
+nlu:
+- intent: greet
+  examples: |
+  - hey
+  - hi
+  - whats up
+```
+
+However, its also possible to use an extended format if you want to include
+metadata for your examples:
+
+```yaml
+nlu:
+- intent: greet
+  examples: 
+  - text: |
+      hi
+    metadata:
+      sentiment: neutral
+  - text: |
+      hey there!
+```
+
+The `metadata` key can contain arbitrary key-value data and the parser will not read its value. 
+Metadata can be useful for storing information relevant to an example, but not used by 
+the NLU pipeline. In the example above, the sentiment of 
+the example is recorded as metadata.
+
+### Entities
+
+Where applicable, [entities])(../glossary#entity) can be annotated in training examples using either the
+abbreviated syntax:
+
+```yaml
+[<entity-text>]("<entity name>")
+```
+
+Or the syntax:
+
+```yaml
+[<entity-text>]{"entity": "<entity name>"}
+```
+
+
+In a training example, this would look like:
+
+```yaml
+nlu:
+- intent: check_balance
+  examples: |
+  - how much do I have on my [savings]("account") account
+  - how much money is in my [checking]{"entity": "account"} account
+  - What's the balance on my [credit card account]{"entity":"account","value":"credit"}
```

Comment:
I think we should always keep this in mind and use this structure throughout our docs:
explanation first, then an example block.

The above has a structure that is more like
explanation, example with additional syntax, explanation of the additional syntax

This makes jumping through sections harder in my opinion. I'd rather remove the third example and introduce it separately in the paragraph that focuses on it.

Reply:
👍 switched it around